packages/expo-face-detector/android/src/main/java/expo/modules/facedetector/ExpoFaceDetector.java

```diff
@@ -1,94 +1,164 @@
-  public static int NO_LANDMARKS = FaceDetector.NO_LANDMARKS;
-  public static int ACCURATE_MODE = FaceDetector.ACCURATE_MODE;
-  public static int FAST_MODE = FaceDetector.FAST_MODE;
+  public static int ALL_CLASSIFICATIONS = FirebaseVisionFaceDetectorOptions.ALL_CLASSIFICATIONS;
+  public static int NO_CLASSIFICATIONS = FirebaseVisionFaceDetectorOptions.NO_CLASSIFICATIONS;
+  public static int ALL_LANDMARKS = FirebaseVisionFaceDetectorOptions.ALL_LANDMARKS;
+  public static int NO_LANDMARKS = FirebaseVisionFaceDetectorOptions.NO_LANDMARKS;
+  public static int ACCURATE_MODE = FirebaseVisionFaceDetectorOptions.ACCURATE;
+  public static int FAST_MODE = FirebaseVisionFaceDetectorOptions.FAST;
+
+  private static final SparseIntArray ORIENTATIONS = new SparseIntArray();
+
+  static {
+    ORIENTATIONS.append(Surface.ROTATION_0, 90);
+    ORIENTATIONS.append(Surface.ROTATION_90, 0);
+    ORIENTATIONS.append(Surface.ROTATION_180, 270);
+    ORIENTATIONS.append(Surface.ROTATION_270, 180);
+  }
+
+  private FirebaseVisionFaceDetector mFaceDetector = null;
 
-  private FaceDetector mFaceDetector = null;
-  private ImageDimensions mPreviousDimensions;
-  private FaceDetector.Builder mBuilder;
+  private Context mContext;
 
   private int mClassificationType = NO_CLASSIFICATIONS;
   private int mLandmarkType = NO_LANDMARKS;
   private float mMinFaceSize = 0.15f;
+  private boolean mTracking = false;
+  private long mMinDetecitonInterval = 0;
+  private long lastDetectionMillis = 0;
   private int mMode = FAST_MODE;
 
   public ExpoFaceDetector(Context context) {
-    mBuilder = new FaceDetector.Builder(context);
-    mBuilder.setMinFaceSize(mMinFaceSize);
-    mBuilder.setMode(mMode);
-    mBuilder.setLandmarkType(mLandmarkType);
-    mBuilder.setClassificationType(mClassificationType);
+    this.mContext = context;
   }
 
   // Public API
 
-  public boolean isOperational() {
+  @Override
+  public void detectFaces(Uri filePath, FacesDetectionCompleted complete, FaceDetectionError error) throws IOException {
+
     if (mFaceDetector == null) {
       createFaceDetector();
     }
-
-    return mFaceDetector.isOperational();
+    FirebaseVisionImage image = FirebaseVisionImage.fromFilePath(mContext, filePath);
+    mFaceDetector.detectInImage(image)
+        .addOnCompleteListener(
+            faceDetectionHandler(FaceDetectorUtils::serializeFace, complete, error)
+        );
   }
 
-  public SparseArray<Face> detect(ExpoFrame frame) {
-    // If the frame has different dimensions, create another face detector.
-    // Otherwise we will get nasty "inconsistent image dimensions" error from detector
-    // and no face will be detected.
-    if (!frame.getDimensions().equals(mPreviousDimensions)) {
-      releaseFaceDetector();
-    }
+  @Override
+  public boolean detectFaces(byte[] imageData, int width, int height, int rotation, boolean mirrored, double scaleX, double scaleY, FacesDetectionCompleted complete, FaceDetectionError error) {
```

Comment:
Yeah... In theory. But in practice that is both undoable and unnecessary.

1. Undoable, because of usage of CameraView library, which hides Camera object from as, thus preventing us from accessing format.
2. Unnecessary, because NV21 is default format for camera preview, unless forcidibly set otherwise. Which we do not do, and we hope CameraView library does not do that either.
Additionally, changing the format in here does not prevent Firebase from detecting faces 🤷‍♂